daffodil-udf/README.md

```diff
@@ -0,0 +1,61 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+# User Defined Functions
+
+## Introduction
+
+Apache Daffodil (incubating) allows execution of Java external/user defined functions in DPath Expressions.
+
+## Getting Started
+
+The implementer will be expected to provide a JAR via the classpath containing at least 2 classes: a provider class and its associated serializable function class(es).
+
+The JAR can have as many provider classes as desired as long as the classes are registered in the *services* folder.
+
+### UDF Implementation
+
+#### Provider Classes
+
+The provider class will be an implementation of Daffodil's `UDFuntionProvider` class. This class must initialize its functionClasses array with all the classes it's providing. It must also implement a lookup function that returns an initialized function class object based on a supplied name and namespace. 
```

Comment:
I think this is out of date now. UDFunctionProvider was changed to UserDefinedFunctionProvider, and the functionClasses array no longer exists.